SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -136,6 +145,110 @@ void FLocalDeploymentManager::RefreshServiceStatus()
 	});
 }
 
+bool FLocalDeploymentManager::CheckIfPortIsBound(int32 Port)
+{
+	ISocketSubsystem* SocketSubsystem = ISocketSubsystem::Get(PLATFORM_SOCKETSUBSYSTEM);
+	bool bSuccess = false;
+
+	// Set our broadcast address
+	TSharedRef<FInternetAddr> BroadcastAddr = SocketSubsystem->CreateInternetAddr();
+	BroadcastAddr->SetBroadcastAddress();
+	BroadcastAddr->SetPort(Port);
+	
+	// Now the listen address
+	TSharedRef<FInternetAddr> ListenAddr = SocketSubsystem->GetLocalBindAddr(*GLog);
+	ListenAddr->SetPort(Port);
+	
+	// Now create and set up our sockets
+	FSocket* ListenSocket = SocketSubsystem->CreateSocket(NAME_Stream, TEXT("Runtime Port Test"), false /* bForceUDP */);
+	if (ListenSocket != nullptr)
+	{
+		ListenSocket->SetReuseAddr();
+		ListenSocket->SetNonBlocking();
+		ListenSocket->SetRecvErr();
+	
+		// Bind to our listen port
+		if (ListenSocket->Bind(*ListenAddr))
+		{
+			bSuccess = ListenSocket->Listen(0 /* MaxBacklog*/ );
+			ListenSocket->Close();
+		}
+		else
+		{
+			UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to bind listen socket to addr (%s) for Runtime Port Test"), *ListenAddr->ToString(true));
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to create listen socket for Runtime Port Test"));
+	}
+
+	// Either we couldn't create the socket or couldn't listen on it so the port is probably bound.
+	return !bSuccess;
+}
+
+bool FLocalDeploymentManager::PreStartCheck()
+{
+	// Check for the known runtime port (5301) which could be blocked.
+	if (CheckIfPortIsBound(RequiredRuntimePort))
+	{
+	
+		// If it exists offer the user the ability to kill it.
+		if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("KillPortBlockingProcess", "A required port is blocked by another process (potentially by an old deployment). Would you like to kill this process?")) == EAppReturnType::Yes)
+		{
+			const FString NetStatCmd = FString::Printf(TEXT("netstat"));
+
+			// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+			FString Args = TEXT("-n -o -a");
```

Comment:
```suggestion
			FString NetStatArgs = TEXT("-n -o -a");
```
Let's keep this consistent with other usages.